src/Container.php

```diff
@@ -64,13 +72,14 @@ public function add($alias, $concrete = null, $singleton = false)
             return null;
         }
 
-        // get a definition of the item
-        $this->items[$alias]['singleton'] = (boolean) $singleton;
-
+        // we need to build a definition
         $factory    = $this->getDefinitionFactory();
         $definition = $factory($alias, $concrete, $this);
 
-        $this->items[$alias]['definition'] = $definition;
+        $this->items[$alias] = [
+            'definition' => $definition,
+            'singleton'  => (boolean) $singleton
```

Comment:
Yep, bool is the correct thing to use: https://github.com/php/php-langspec/blob/0f9985ba76ab7890a9a8440f8a4344c58f0a6761/spec/05-types.md#the-boolean-type.